tests/cursor/bug0920-001.phpt

```diff
@@ -0,0 +1,24 @@
+--TEST--
+PHPC-924: Using a projection with an empty field name causes a crash when destroying cursor
+--SKIPIF--
+<?php require __DIR__ . "/../utils/basic-skipif.inc"; CLEANUP(STANDALONE) ?>
+--FILE--
+<?php
+require_once __DIR__ . "/../utils/basic.inc";
+
+$manager = new MongoDB\Driver\Manager(STANDALONE);
+
+$query = new MongoDB\Driver\Query([], ['projection' => ['' => 1]]);
+
+echo throws(function() use ($manager, $query) {
+    $cursor = $manager->executeQuery(NS, $query);
```

Comment:
With libmongoc 1.5.4, the segfault occurs here when the cursor is checked for an error, an exception is thrown, and the cursor is destroyed. In 1.5.5, the segfault is fixed but we still throw the exception here.

I think it might make more sense to do the same BSON validation that libmongoc does during cursor initialization in our own Query constructor, so we can instead throw an InvalidArgumentException earlier. This duplicates the validation logic, but the filter and options documents should be small enough where this isn't a concern. If this sound reasonable, I'll open a new ticket to address this for 1.3.0.